presto-hive/src/test/java/io/prestosql/plugin/hive/TestShowStats.java

```diff
@@ -212,62 +256,145 @@ public void testShowStatsForNonExistingColumnFails()
+        assertQuery("SHOW STATS FOR (SELECT 1 FROM nation_partitioned)",
+                "SELECT * FROM (VALUES " +
+                        "   ('_col0', null, 1, 0, null, 1, 1), " +
+                        "   (null, null, null, null, 25, null, null))");
     }
 
     @Test
-    public void testShowStatsForAliasedColumnQueryFails()
+    public void testShowStatsForAliasedColumnQuery()
     {
-        assertQueryFails("SHOW STATS FOR (SELECT nationkey, name, name AS name2 FROM nation_partitioned WHERE regionkey > 0 and regionkey < 4)", ".*Column aliasing is not supported in SHOW STATS SELECT clause");
+        assertQuery("SHOW STATS FOR (SELECT nationkey, name, name AS name2 FROM nation_partitioned WHERE regionkey > 0 and regionkey < 4)",
+                "SELECT * FROM (VALUES " +
+                        "   ('nationkey', null, 5, 0, null, 1, 24), " +
+                        "   ('name', 109, 5, 0, null, null, null), " +
+                        "   ('name2', 109, 5, 0, null, null, null), " +
+                        "   (null, null, null, null, 15, null, null))");
     }
 
     @Test
-    public void testShowStatsForNonIdentifierColumnFails()
+    public void testShowStatsForNonIdentifierColumn()
     {
-        assertQueryFails("SHOW STATS FOR (SELECT CONCAT('some', 'value') FROM nation_partitioned)", ".*Only table columns names are supported in SHOW STATS SELECT clause");
+        assertQuery("SHOW STATS FOR (SELECT CONCAT('some', 'value') FROM nation_partitioned)",
+                "SELECT * FROM (VALUES " +
+                        "   ('_col0', null, 1, 0, null, null, null), " +
+                        "   (null, null, null, null, 25, null, null))");
     }
 
     @Test
-    public void testShowStatsForColumnExpressionFails()
+    public void testShowStatsForColumnExpression()
     {
-        assertQueryFails("SHOW STATS FOR (SELECT nationkey + 1 FROM nation_partitioned)", ".*Only table columns names are supported in SHOW STATS SELECT clause");
+        assertQuery("SHOW STATS FOR (SELECT nationkey + 1 FROM nation_partitioned)",
+                "SELECT * FROM (VALUES " +
+                        "   ('_col0', null, 5, 0, null, 1, 25), " +
+                        "   (null, null, null, null, 25, null, null))");
     }
 
     @Test
-    public void testShowStatsWithGroupByFails()
+    public void testShowStatsWithGroupBy()
     {
-        assertQueryFails("SHOW STATS FOR (SELECT avg(totalprice) FROM orders GROUP BY orderkey)", ".*Only table columns names are supported in SHOW STATS SELECT clause");
+        assertQuery("SHOW STATS FOR (SELECT avg(totalprice) FROM orders GROUP BY orderkey)",
+                "SELECT * FROM (VALUES " +
+                        "   ('_col0', null, null, null, null, null, null), " +
+                        "   (null, null, null, null, null, null, null))");
```

Comment:
We should know the row count since it's NDV(orderkey), which is known.
Is it unknown because of partial aggregations? 
Please file an issue to further improve this.